app/src/main/java/org/mozilla/fenix/customtabs/CustomTabToolbarMenu.kt

```diff
@@ -99,60 +99,58 @@ class CustomTabToolbarMenu(
+        label = context.getString(R.string.browser_menu_share),
+        imageResource = R.drawable.mozac_ic_share,
+        textColorResource = primaryTextColor(),
+        iconTintColorResource = primaryTextColor()
+    ) {
+        onItemTapped.invoke(ToolbarMenu.Item.Share)
+    }
 
-            BrowserMenuImageText(
-                context.getString(R.string.browser_menu_find_in_page),
-                R.drawable.mozac_ic_search,
-                ThemeManager.resolveAttribute(R.attr.primaryText, context)
-            ) {
-                onItemTapped.invoke(ToolbarMenu.Item.FindInPage)
-            },
+    private val desktopMode = BrowserMenuSwitch(
+        label = context.getString(R.string.browser_menu_desktop_site),
+        initialState = { session?.desktopMode ?: false }, listener = { checked ->
+            onItemTapped.invoke(ToolbarMenu.Item.RequestDesktop(checked))
+        })
+
+    private val findInPage = BrowserMenuImageText(
+        label = context.getString(R.string.browser_menu_find_in_page),
+        imageResource = R.drawable.mozac_ic_search,
+        iconTintColorResource = primaryTextColor()
+    ) {
+        onItemTapped.invoke(ToolbarMenu.Item.FindInPage)
+    }
 
-            SimpleBrowserMenuItem(
-                {
-                    val appName = context.getString(R.string.app_name)
-                    context.getString(R.string.browser_menu_open_in_fenix, appName)
-                }(),
-                textColorResource = ThemeManager.resolveAttribute(
-                    R.attr.primaryText,
-                    context
-                )
-            ) {
-                onItemTapped.invoke(ToolbarMenu.Item.OpenInFenix)
-            },
+    private val openInFenix = SimpleBrowserMenuItem(
+        label = {
+            val appName = context.getString(R.string.app_name)
+            context.getString(R.string.browser_menu_open_in_fenix, appName)
+        }(),
+        textSize = ToolbarMenu.CAPTION_TEXT_SIZE,
+        textColorResource = primaryTextColor()
+    ) {
+        onItemTapped.invoke(ToolbarMenu.Item.OpenInFenix)
+    }
 
-            BrowserMenuDivider(),
+    private val poweredBy = SimpleBrowserMenuItem(
+        label = {
+            val appName = context.getString(R.string.app_name)
+            context.getString(R.string.browser_menu_powered_by, appName).toUpperCase()
+        }(),
+        textSize = ToolbarMenu.CAPTION_TEXT_SIZE,
+        textColorResource = primaryTextColor()
+    )
 
-            SimpleBrowserMenuItem(
-                {
-                    val appName = context.getString(R.string.app_name)
-                    context.getString(R.string.browser_menu_powered_by2, appName).toUpperCase()
-                }(),
-                ToolbarMenu.CAPTION_TEXT_SIZE,
-                ThemeManager.resolveAttribute(R.attr.primaryText, context)
-            )
-        )
-    }
+    private fun primaryTextColor() = ThemeManager.resolveAttribute(R.attr.primaryText, context)
```

Comment:
nit: Is it worth putting this in the `ThemeManager` itself 🤔 other places in our code base resolve this attr and if it's frequently used it may be worth it 😄

Reply:
Good idea, but since we need to pass context anyway it winds up not really saving us much, so I'm going to leave it as is.
```
class ThemeManager {
...
  fun primaryTextColor(context: Context): Int = 
      ThemeManager.resolveAttribute(R.attr.primaryText, context)
}
```